applications/serial_lte_modem/src/slm_at_host.c

```diff
@@ -1032,7 +1033,7 @@ static void uart_callback(const struct device *dev, struct uart_event *evt, void
 		break;
 	case UART_RX_DISABLED:
 		LOG_DBG("RX_DISABLED");
-		k_sem_give(&rx_change);
+		uart_state = STATE_DISABLED;
 
 		if (slm_operation_mode == SLM_DATA_MODE) {
 			datamode_rx_disabled = true;
```

Comment:
This flag `datamode_rx_disabled` is redundant now, can use `uart_state` instead

Reply:
Based on tests failing in CI and @junqingzou:s comments. Changing approach.